keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -259,66 +248,86 @@ the API:
   forwarding requests or persisting data) cannot be undone, but no future
   actions should be allowed.
 
-The examples below include information about what happens when the ReferenceGrant
-is removed as data points.
-
 ### Example Usage
 
-#### Gateway API Gateway Referencing Secret
+#### Gateway API Cross-Namespace Secret Reference
 
-In this example (from the Gateway API docs), we have a Gateway in the
-`gateway-api-example-ns1` namespace, referencing a Secret in the
-`gateway-api-example-ns2` namespace. The following ReferenceGrant allows this:
+Authors of APIs can bundle ClusterReferenceGrants with their API definitions
+for any references they expect to be common. For example, Gateway API could
+bundle the following ClusterReferenceGrant to authorize same-namespace
+references from Gateways to Secrets for "tls".
 
 ```yaml
-apiVersion: gateway.networking.k8s.io/v1beta1
-kind: Gateway
+kind: ClusterReferenceGrant
+apiVersion: reference.authorization.k8s.io/v1alpha1
 metadata:
-  name: cross-namespace-tls-gateway
-  namespace: gateway-api-example-ns1
-spec:
-  gatewayClassName: acme-lb
-  listeners:
-  - name: https
-    protocol: HTTPS
-    port: 443
-    hostname: "*.example.com"
-    tls:
-      # There's a Kind/Resource mismatch here, which sucks, but it is not
-      # easily fixable, since Gateway is already a beta, close to GA
-      # object.
-      certificateRefs:
-      - kind: Secret
-        group: ""
-        name: wildcard-example-com-cert
-        namespace: gateway-api-example-ns2
----
-apiVersion: authorization.k8s.io/v1alpha1
+  name: gateway-tls
+from:
+  group: gateway.networking.k8s.io
+  resource: gateways
+  version: v1
+  path: ".spec.listeners[*].tls.certificateRefs[*]"
+to:
+  group: ""
+  resource: secrets
+for: tls
+```
+
+Implementations of APIs can bundle ClusterReferenceConsumer resources as part of
+their deployment. This resource links a subject (likely the ServiceAccount
+used by the controller) to matching (Cluster)ReferenceGrants. The subject of
+this ClusterReferenceConsumer will receive authorization for any
+(Cluster)ReferenceGrant with matching `from`, `to`, and `for` values.
+
+```yaml
+kind: ClusterReferenceConsumer
+apiVersion: reference.authorization.k8s.io/v1alpha1
+metadata:
+  name: contour-gateway
+subject:
+  kind: ServiceAccount
+  name: contour
+  namespace: contour-system
+from:
+  group: gateway.networking.k8s.io
+  resource: gateways
+to:
+  group: ""
+  resource: secrets
+for: tls
+```
+
+Finally, users can use ReferenceGrant to authorize specific cross-namespace
+references. For example, the following ReferenceGrant authorizes references from
+Gateways in the `prod` namespace to the `acme-tls` Secret in the `prod-tls`
+namespace:
+
+```yaml
 kind: ReferenceGrant
+apiVersion: reference.authorization.k8s.io/v1alpha1
 metadata:
```

Comment:
I think that it would be desirable to add "namespace: prod-tls" to the metadata section.